runtime/near-vm-logic/src/gas_counter.rs

```diff
@@ -16,12 +16,7 @@ thread_local! {
 
 #[cfg(all(feature = "costs_counting", feature = "protocol_feature_evm"))]
 pub fn reset_evm_gas_counter() -> u64 {
-    let mut ret = 0;
-    EVM_GAS_COUNTER.with(|f| {
-        ret = *f.borrow();
-        *f.borrow_mut() = 0;
-    });
-    ret
+    EVM_GAS_COUNTER.with(|f| f.replace(0))
```

Comment:
[Jones's trick](http://giphygifs.s3.amazonaws.com/media/MS0fQBmGGMaRy/giphy.gif) can simplify a lot of things. Many types have `.replace` method: https://doc.rust-lang.org/stable/std/?search=replace. There's also `std::mem::replace`.